EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -13,13 +14,33 @@ namespace EngineLayer
 {
     public static class PEP_Analysis
     {
+        private static readonly string filepathSubstitue = "SubstitueFilePath";//this is only for unit test that have psms with no filename.
+
         public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch> psms)
         {
-            Dictionary<string, int> accessionAppearances = GetAccessionCounts(psms);
+            string searchType = DetermineSearchType(psms);
+
+            //These two dictionaries contain the average and standard deviations of hydrophobicitys measured in 1 minute increments accross each raw
+            //file separately. An individully measured hydrobophicty calculated for a specific PSM sequence is compared to these values by computing
+            //the z-score. That z-score is used in the the machine learning.
+            Dictionary<string, Dictionary<int, double>> RetentionTimeHydrophobicityDeviation = new Dictionary<string, Dictionary<int, double>>();
+            Dictionary<string, Dictionary<int, double>> RetentionTimeHydrophobicityAverage = new Dictionary<string, Dictionary<int, double>>();
```

Comment:
I think that a Dictionary<string, Dictionary<int, Tuple<double, double>>> may work to simplify things. Having item 1 as the deviation and item 2 as the average or vice versa. That way all of the information is stored in a single dictionary for unmodified and a single place for modified peptides. Then for calculation of Z-score you just have to pass in this single dictionary instead of both. Just a thought.

Reply:
switched to tuple now